syntax = "proto3";

package health;

// The health check service definition.
service Health {
  // Sends a health check request.
  rpc Check(HealthCheckRequest) returns (HealthCheckResponse) {}
}

// The request message containing the service name.
message HealthCheckRequest {
  string service = 1;
}

// The response message containing the status.
message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
  }
  ServingStatus status = 1;
}